pol-core/plib/uoexpansion.h

```diff
@@ -77,77 +130,87 @@ enum class ExpansionVersion : u8
   KR,
   SA,
   HSA,
-
-  LastVersion = HSA
+  TOL,
+  LastVersion = TOL
 };
-const int numExpansions = static_cast<int>( ExpansionVersion::LastVersion ) + 1;
-const char* getExpansionName( ExpansionVersion x );
-
+std::string getExpansionName( ExpansionVersion x );
+ExpansionVersion getExpansionVersion( const std::string& str );
+B9Feature getDefaultExpansionFlag( ExpansionVersion x );
 
-class UOExpansion
+enum class FaceSupport : u8
 {
-public:
-  virtual ~UOExpansion() = default;
-  virtual bool hasFeature( A9Feature feature ) const = 0;
-  virtual bool hasFeature( B9Feature feature ) const = 0;
-  virtual u32 A9Flags() const = 0;
-  virtual u32 B9Flags() const = 0;
-
-  virtual ExpansionVersion version() const { return ExpansionVersion::T2A; }
-  virtual int characterSlots() const { return 5; }
+  None,
+  Basic,
+  RolePlay
 };
 
-class ClientFeatures
+// hold in server
+class ServerFeatures
 {
-  const UOExpansion& m_expansion;
-
 public:
-  ClientFeatures( UOExpansion& expansion ) : m_expansion( expansion ) {}
-  bool hasCustomHousing() const { return supports( ExpansionVersion::AOS ); }
-  bool hasSamuraiNinja() const { return m_expansion.hasFeature( A9Feature::SamuraiNinja ); }
-  bool hasElvenRace() const { return m_expansion.hasFeature( A9Feature::ElvenRace ); }
-  bool hasContextMenus() const { return m_expansion.hasFeature( A9Feature::ContextMenus ); }
-  bool hasNewMovementPacket() const { return m_expansion.hasFeature( A9Feature::NewMovement ); }
-  // Commonly used expansion
-  bool supportsAOS() const { return supports( ExpansionVersion::AOS ); }
-  bool supports( ExpansionVersion version ) const
+  ExpansionVersion Expansion() const { return expansion; };
```

Comment:
My spidey-sense is tingling with the naming of this ...
```cpp
ExpansionVersion Expansion();
FaceSupport faceSupport();
```

Nit to rename to `expansion()` or `expansion_version()` 🤷